### PR TITLE
add gas refund handler

### PR DESCRIPTION
**Pull request by AaronZgl**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by KamiD — CHANGES_REQUESTED** [review]
gas refund only works on evm tx

**Review by summerpro — DISMISSED** [review]

**Comment by AaronZgl** [comment]
> gas refund only works on evm tx

The cosmos gas refund function has been removed.

File: app/refund/refund.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		case evmtypes.MsgEthereumTx:
+			gasRefundHandler = NewGasRefundDecorator(ak, sk)
+		default:
+			return nil
+		}
+		return gasRefundHandler(ctx, tx)
+	}
+}
+
+type GasRefundHandler struct {
```
Comment (xiangjianmeng) [inline app/refund/refund.go]: may consider `GasRefund`

File: app/refund/refund.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	ak           keeper.AccountKeeper
+	supplyKeeper types.SupplyKeeper
+}
+
+func (grh GasRefundHandler) GasRefund(ctx sdk.Context, tx sdk.Tx) (err error) {
+
+	currentGasMeter := ctx.GasMeter()
+	TempGasMeter := sdk.NewInfiniteGasMeter()
+	ctx = ctx.WithGasMeter(TempGasMeter)
+
```
Comment (xiangjianmeng) [inline app/refund/refund.go]: may consider `Refund`

**Review by xiangjianmeng — DISMISSED** [review]

**Review by xiangjianmeng — DISMISSED** [review]

**Review by summerpro — DISMISSED** [review]

**Review by xiangjianmeng — DISMISSED** [review]

**Review by KamiD — DISMISSED** [review]

**Review by KamiD — APPROVED** [review]

**Review by xiangjianmeng — APPROVED** [review]

**Review by summerpro — APPROVED** [review]

**Pull request merged**